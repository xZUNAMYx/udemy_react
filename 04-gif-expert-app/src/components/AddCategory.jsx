import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event)=>{
        setInputValue (event.target.value);
    }
  
    const onSumit = (event)=>{
        event.preventDefault();
        if(inputValue.trim().length <= 1) return; 

        // setCategories((categories)=>[...categories, inputValue]);
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={(event)=> onSumit(event)} aria-label="form">
            <input 
                type="text" 
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ (event)=>onInputChange(event) }
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}