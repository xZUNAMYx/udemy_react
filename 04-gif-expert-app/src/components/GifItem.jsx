import ProtoTypes from "prop-types";

export const GifItem = ({title, url, id}) => {
  return (
        // <li key={image.id}>{image.title}</li>
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
  );
}

GifItem.protoTypes = {
  title: ProtoTypes.string.isRequired,
  url: ProtoTypes.string.isRequired,
}

GifItem.defaultProps ={
  title: 'No hay título',
  url: 'no se encontró alguna url',
}
