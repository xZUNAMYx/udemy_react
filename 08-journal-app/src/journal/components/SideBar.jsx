import { TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"

export const SideBar = ({ drawerWidth = 240 }) => {
    return (
        <Box
            component='nav'
            sx={{ width: { sm: drawerWidth }, flexShrink: {sm: 0 }}}
        >
            <Drawer
                variant="permanent" // temporary
                open={true}
                sx={{
                    display: { xs:'block'},
                    '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
                }}
            >
                <Toolbar>
                    <Typography variant="h6" noWrap component='div'>
                        Andres Alvarez
                    </Typography>
                </Toolbar>
                <Divider />

                <List>
                    {
                        ['Enero', 'Febrero'].map( text => (
                            <ListItem key={ text } disablePadding> 
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot />
                                    </ListItemIcon>
                                    <Grid container>
                                        <ListItemText primary={ text } />
                                        <ListItemText secondary={ 'Otro texto' } />
                                    </Grid>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                </List>
        
            </Drawer>
        </Box>
    )
}
