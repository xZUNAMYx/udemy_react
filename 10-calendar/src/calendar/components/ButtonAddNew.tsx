import { addHours } from "date-fns";
import { useUiStore } from '../../hooks/useUiStore';
import { useCalendarStore } from "../../hooks/useCalendarStore";


export const ButtonAddNew = () => { 

    const { openDateModal } = useUiStore();
    const { setActiveEvent } = useCalendarStore();

    const handleClickNew = () =>{
        setActiveEvent({
            title: '',
            notes: '',
            start: new Date(),
            end: addHours( new Date(), 2),
        }),

        openDateModal();
    }

    return (
    <button
        className="btn btn-primary fab"
        onClick={ handleClickNew }
    >
        <i className="fas fa-plus"></i>

    </button>
    )
}
