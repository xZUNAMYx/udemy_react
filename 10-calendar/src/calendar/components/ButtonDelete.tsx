import { useCalendarStore } from "../../hooks/useCalendarStore";


export const ButtonDelete = () => { 

    const { startDeletingEvent, hasEventSelected } = useCalendarStore();

    const handleDelate = () =>{
        startDeletingEvent();
    }

    return (
    <button
        className="btn btn-danger fab-delete"
        onClick={ handleDelate }
        style={{
            display: hasEventSelected ? '' : 'none'
        }}
    >
        <i className="fas fa-trash-alt"></i>

    </button>
    )
}
