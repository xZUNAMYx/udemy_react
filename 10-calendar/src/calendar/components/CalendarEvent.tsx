interface Event {
    user: {
        _id: string,
        name: string,
    };
    title: string;  
}
 
interface Props {
    event: Event
}

export const CalendarEvent = ( { event } : Props ) => {
    const { title } = event;
    // const { title, user} = event;

    return (
        <>
            <strong>{ title }</strong>
            {/* <strong> - { user.name }</strong> */}
        </>
    );
}
