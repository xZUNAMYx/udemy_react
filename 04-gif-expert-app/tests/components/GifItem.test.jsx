import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Prueba en GifItem.jsx', () => { 
    const title = 'Primera prueba';
    const url = 'https://google.com/fotos-de-gatitos'

    test('Debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URl y el ALT indicado', () => {
        render(<GifItem title={title} url={url}/>);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });
    
    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render(<GifItem title={title} url={url}/>);
        expect( screen.getByText( title )).toBeTruthy();
    });
    
    
})