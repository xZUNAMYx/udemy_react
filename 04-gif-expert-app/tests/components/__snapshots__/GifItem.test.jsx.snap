// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem.jsx Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Primera prueba"
      src="https://google.com/fotos-de-gatitos"
    />
    <p>
      Primera prueba
    </p>
  </div>
</div>
`;
